cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeClientService.java

```diff
@@ -96,8 +117,29 @@ public class RuntimeClientService extends AbstractRetryableScheduledService {
     this.runtimeClient = runtimeClient;
     this.fetchLimit = cConf.getInt(Constants.RuntimeMonitor.BATCH_SIZE);
     this.programFinishTime = new AtomicLong(-1L);
+
+    // Initialize encoders and decoders for metrics aggregation.
+    Schema schema = null;
```

Comment:
Please refactor the logic of aggregating metrics into a separate class to keep the RuntimeClientService cleaner.

Reply:
Done.